### PR TITLE
Add CrashlyticsReport JSON de/serialization

### PR DESCRIPTION
**Pull request by mrwillis21**
Uses firebase-encoders-json DataEncoder for serialization and
uses JsonReader manually for deserialization, as there is not yet
a firebase-decoders-json.

### CODE REVIEW

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/serialization/CrashlyticsReportJsonTransform.java [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+    return CRASHLYTICS_REPORT_JSON_ENCODER.encode(event);
+  }
+
+  @Nullable
+  public CrashlyticsReport reportFromJson(@NonNull String json) {
+    try (JsonReader jsonReader = new JsonReader(new StringReader(json))) {
+      return parseReport(jsonReader);
+    } catch (IOException e) {
+      // TODO: Handle the IOException
+      return null;
```
Comment (bklimt) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/serialization/CrashlyticsReportJsonTransform.java]: I don't love this, but I won't push back too hard on it. It's just harder to make sure you deal with null upstream than to deal with an Exception.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/serialization/CrashlyticsReportJsonTransform.java [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+          jsonReader.skipValue();
+          break;
+      }
+    }
+    jsonReader.endObject();
+
+    return builder.build();
+  }
+
+  private static CrashlyticsReport.Session.OperatingSystem parseOs(JsonReader jsonReader) throws IOException {
```
Comment (bklimt) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/serialization/CrashlyticsReportJsonTransform.java]: "Os" you're killing me. I'd argue that in a line with `IOException`, we should be able to call this `parseOS`. :-)

But I guess you want to match it match the name in the object, so whatever.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/serialization/CrashlyticsReportJsonTransform.java (hunk 1 above)
Comment (mrwillis21) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/serialization/CrashlyticsReportJsonTransform.java]: Agreed, hence the TODO.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/serialization/CrashlyticsReportJsonTransform.java (hunk 2 above)
Comment (mrwillis21) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/serialization/CrashlyticsReportJsonTransform.java]: Yeah. :man_shrugging:

**Pull request merged**